arm-datalake-analytics/account/2016-11-01/swagger/account.json

```diff
@@ -1573,6 +1563,29 @@
         "location"
       ],
       "x-ms-azure-resource": true
+    },
+    "SubResource": {
+      "description": "The Sub Resource model definition.",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource Id"
+        },
+        "name": {
```

Comment:
It is actually correct, since the only time this object is created by a user is during ADLA account creation, when they want to specify a list of these accounts to include during initial ADLA Create. Otherwise, during "add" operations they use just the AddStorageAccountParameters object.